daily_jobs.py

```diff
@@ -0,0 +1,202 @@
+        talks
+    ))
+
+    if not any([this_week_talks, next_week_talks]):
+        # Nothing to announce
+        return
+
+    message = WEEKLY_ANNOUNCEMENT_TEMPLATE.render(
+        this_week_talks=this_week_talks,
+        next_week_talks=next_week_talks,
+    )
+    data = {
+        "from": "VSF team <no-reply@mail.virtualscienceforum.org>",
+        "to": "speakers_corner@mail.virtualscienceforum.org",
+        "subject": "Speakers' Corner weekly schedule",
+        "text": common.markdown_to_plain(message),
+        "html": common.markdown_to_email(message),
+    }
+
+    response = common.api_query(
+        requests.post,
+        common.MAILGUN_DOMAIN + "messages",
+        data=data
+    )
+    logging.info("Sent the weekly update.")
+    return response
+
+
+def add_zoom_registrants_to_mailgun():
+    """Add registrants from Zoom to Mailgun mailing list."""
+    # Get all meetings
+    meetings = common.all_meetings("me")
+
+    # Filter by past meetings
+    past_lrc = [
+        meeting for meeting in meetings
+        if ("Long Range" in meeting['topic']
+            and pandas.to_datetime(meeting['start_time']) < datetime.datetime.now(tz=pytz.UTC)
+           )
+    ]
+
+    # Restrict to past 20 meetings (not likely to have more than 20 in a day)
+    past_lrc = past_lrc[-20:]
+
+    for lrc in past_lrc:
+        # Get registrants
+        lrc_registrants = common.meeting_registrants(lrc['id'])
+
+        # Filter those who want to sign up for emails
+        member_data = dict(members=json.dumps([
+            dict(address=i['email'], name="{0} {1}".format(i.get('first_name'), i.get('last_name','')))
+            for i in lrc_registrants if (i.get('May we contact you about future Virtual Science Forum events?','') == "Yes")
+        ]))
+
+        api_query(
+            post, f'lists/{announce_list}/members.json',
+            data=member_data
+        )
+
+    return
+
+if __name__ == "__main__":
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    now = datetime.datetime.now(tz=pytz.UTC)
+    exceptions = common.CollectExceptions()
+
+    talks, _ = common.talks_data()
+    talks = [talk for talk in talks if talk["event_type"] == "speakers_corner"]
+    logging.info(f"Loaded {len(talks)} talks.")
+    for talk in talks:
+        talk["time"] = talk["time"].replace(tzinfo=pytz.UTC)
+
+    # Weekly emails sent on Sundays
+    if now.weekday() == 6:
+        with exceptions:
+            weekly_speakers_corner_update(talks)
+            logging.info(f"Sent a weekly Speakers' corner announcement")
+
+    # Email the speaker their video link for a talk that took place 10h ago
+    # In principle we could be faster, but this is to guarantee that the
+    # Transcription finished.
+    with exceptions:
+        finished_talk = next(
+            (
+                talk for talk in talks
+                if (now - talk["time"]).total_seconds() // 3600 == 9
```

Comment:
This is a too strict check. Should instead check if the talk happened within the last day (giving some grace period for the recording to finish processing).

Reply:
So check if happened [now - 32hrs, now - 8hrs] window?

Reply:
Something like that. Let's check for round hours though, to avoid a problem with scheduling. Also the video is usually available already within an hour after the meeting, so let's push the lower border to 2 hours.